### ISSUE TITLE
Webcamize says gphoto2 didn't detect any cameras, but it did

### ISSUE THREAD

**Issue by gleporeNARA** [weebney/webcamize#3]
upstairs:~/Downloads/motion$ webcamize
/usr/local/bin/webcamize: gphoto2 didn't detect any cameras
upstairs:~/Downloads/motion$ gphoto2 --summary
Camera summary:                                                                
Manufacturer: Nikon Corporation
Model: D3100
  Version: V1.01
  Serial Number: 3301069
Vendor Extension ID: 0xa (1.0)
Vendor Extension Description: microsoft.com: 1.0

Capture Formats: JPEG Undefined Type
Display Formats: JPEG, Undefined Type, Association/Directory, DPOF, Script, Apple Quicktime

Device Capabilities:
        File Download, File Deletion, File Upload
        Generic Image Capture, No Open Capture, No vendor specific capture

Storage Devices Summary:
store_00010001:
        StorageDescription: 
        VolumeLabel: NIKON D3100
        Storage Type: Removable RAM (memory card)
        Filesystemtype: Digital Camera Layout (DCIM)
        Access Capability: Read Only with Object deletion
        Maximum Capability: 15923150848 (15185 MB)
        Free Space (Bytes): 15918530560 (15181 MB)
        Free Space (Images): 1854

Device Property Summary:
Battery Level(0x5001):(read only) (type=0x2) Range [0 - 100, step 1] value: 100% (100)
Image Size(0x5003):(readwrite) (type=0xffff) Enumeration [
        '4608x3072',
        '3456x2304',
        '2304x1536'
        ] value: '4608x3072'
Compression Setting(0x5004):(readwrite) (type=0x2) Enumeration [0,1,2,4,5] value: JPEG Fine (2)
White Balance(0x5005):(readwrite) (type=0x4) Enumeration [2,4,5,6,7,32784,32785,32787] value: Automatic (2)
F-Number(0x5007):(read only) (type=0x4) Enumeration [400,450,500,560,630,710,800,900,1000,1100,1300,1400,1600,1800,2000,2200,2500] value: f/4 (400)
Focal Length(0x5008):(read only) (type=0x6) Range [1800 - 5500, step 1] value: 24 mm (2400)
Focus Mode(0x500a):(read only) (type=0x4) Enumeration [1,32784,32785,32786,32787] value: AF-A (32786)
Exposure Metering Mode(0x500b):(readwrite) (type=0x4) Enumeration [2,3,4] value: Multi-spot (3)
Flash Mode(0x500c):(readwrite) (type=0x4) Enumeration [2,4,32784] value: Auto (32784)
Exposure Time(0x500d):(read only) (type=0x6) Enumeration [50,62,80,100,125,166,200,250,333,400,500,666,769,1000,1250,1666,2000,2500,3333,4000,5000] value: 0.0017 sec (166)
Exposure Program Mode(0x500e):(read only) (type=0x4) Enumeration [1,2,3,4,32784,32785,32786,32787,32788,32789,32790,32791] value: Auto (32784)
Exposure Index (film speed ISO)(0x500f):(readwrite) (type=0x4) Enumeration [100,200,400,800,1600,3200,6400,12800] value: ISO 200 (200)
Exposure Bias Compensation(0x5010):(readwrite) (type=0x3) Enumeration [-5000,-4666,-4333,-4000,-3666,-3333,-3000,-2666,-2333,-2000,-1666,-1333,-1000,-666,-333,0,333,666,1000,1333,1666,2000,2333,2666,3000,3333,3666,4000,4333,4666,5000] value: 0.0 stops (0)
Date & Time(0x5011):(readwrite) (type=0xffff) '20230808T101105'
Still Capture Mode(0x5013):(read only) (type=0x4) Enumeration [1,2,32785,32790] value: Single Shot (1)
Burst Number(0x5018):(readwrite) (type=0x4) Range [1 - 99, step 1] value: 1
Focus Metering Mode(0x501c):(readwrite) (type=0x4) Enumeration [2,32784,32785,32786] value: Closest Subject (32785)
UseDeviceStageFlag(0xd303):(read only) (type=0x2) 1
Property 0xd406:(readwrite) (type=0xffff) 'Windows/6.0.5330.0 MTPClassDriver/6.0.5330.0'
Property 0xd407:(read only) (type=0x6) 1

**Comment by weebney** [comment]
Can you give me some more details?

What camera model, distro, etc. are you using?

Also, can you show me the outputs of `gphoto2 --autodetect` as well as `gphoto2 --summary`?

**Comment by gleporeNARA** [comment]
Thanks for fixing the bug! Unfortunately it looks like the Nikon D3100 doesn't support LiveView, which probably means it can't be used as a webcam.

**Comment by weebney** [comment]
Ah, that sucks! I was just thinking "wow, a modern camera that isn't supported?" because I literally remember the D3100 coming out what felt like really not that long ago—what I didn't realize is that it's been out for 13 years already...

**Comment by gleporeNARA** [comment]
Yeah, I got the camera second hand about five years ago. I'm just trying to come up with a better solution than using a crappy $50 webcam. I was also unable to get my phone (Samsung Galaxy A50) or an older phone (Galaxy Note 8) to work.  Got 'Sorry, your camera does not support generic capture' for both. Looks like the problem is with Gphoto2...

**Comment by weebney** [comment]
Seems to be more of an exception than a rule that smartphones support the PTP features required for live casting, which is weird because afaik android's implementation of PTP supports live view by default. Probably disabled on most phones for security reasons, but oh well. Could be worth buying a second hand Canon Powershot off of eBay, as most of them seem to support live view. 

I'm going to add a list of supported devices in a few minutes when I get home from lunch.

**Comment by weebney** [comment]
You can check out that list here: https://github.com/weebney/webcamize/blob/master/assets/supported.md
